fuel-tx/src/transaction/consensus_parameters/gas.rs

```diff
@@ -584,22 +595,66 @@ impl GasCostsValues {
 impl DependentCost {
     /// Create costs that are all set to zero.
     pub fn free() -> Self {
-        Self {
+        Self::HeavyOperation {
             base: 0,
-            dep_per_unit: 0,
+            gas_per_unit: 0,
         }
     }
 
     /// Create costs that are all set to one.
     pub fn unit() -> Self {
-        Self {
+        Self::HeavyOperation {
             base: 1,
-            dep_per_unit: 0,
+            gas_per_unit: 1,
```

Comment:
I think the idea is that opcode always has a cost `1`=D

```suggestion
            gas_per_unit: 0,
```